### PR TITLE
[5814] Create Group form allowing to create a group with empty space as name

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5814

### CODE REVIEW

File: ui/app/src/components/CreateUserDialog/CreateUserDialogContainer.tsx [hunk 1]
```diff
@@ -116,6 +124,13 @@ const useStyles = makeStyles()((theme) => ({
   }
 }));
 
+const translations = defineMessages({
+  invalidMinLength: {
+    id: 'createUserDialog.invalidMinLength',
+    defaultMessage: 'Min {length} characters.'
```
Comment (rart) [inline ui/app/src/components/CreateUserDialog/CreateUserDialogContainer.tsx]:
```suggestion
    defaultMessage: 'Min {length} characters'
```

File: ui/app/src/components/CreateUserDialog/CreateUserDialogContainer.tsx [hunk 2]
```diff
@@ -148,7 +163,12 @@ export function CreateUserDialogContainer(props: CreateUserDialogContainerProps)
       });
       setSubmitted(true);
       if (Object.values(newUser).every(Boolean)) {
-        create(newUser)
+        const trimmedNewUser = {};
+        Object.entries(newUser).forEach(([key, value]) => {
+          trimmedNewUser[key] = typeof value === 'boolean' ? value : value.trim();
```
Comment (rart) [inline ui/app/src/components/CreateUserDialog/CreateUserDialogContainer.tsx]:
```suggestion
          trimmedNewUser[key] = typeof value === 'string' ? value.trim() : value;
```

File: ui/app/src/components/EditUserDialog/utils.ts [hunk 3]
```diff
@@ -54,4 +55,7 @@ export interface EditUserDialogUIProps {
   onCloseResetPasswordDialog(): void;
   onDelete(username: string): void;
   onResetPassword(value: boolean): void;
+  isInvalidEmail(email: string): boolean;
+  validateRequiredField(field: string): boolean;
+  validateFieldMinLength(key: string, value: string): boolean;
```
Comment (rart) [inline ui/app/src/components/EditUserDialog/utils.ts]: Can we stop passing utils as props?

File: ui/app/src/components/EditUserDialog/EditUserDialogUI.tsx [hunk 4]
```diff
@@ -207,11 +240,19 @@ export function EditUserDialogUI(props: EditUserDialogUIProps) {
...
                       id="email"
                       onChange={(e) => onInputChange({ email: e.currentTarget.value })}
                       value={user.email}
+                      error={validateRequiredField(user.email) || isInvalidEmail(user.email)}
                       fullWidth
+                      helperText={
+                        validateRequiredField(user.email) ? (
+                          <FormattedMessage id="editUserDialog.emailRequired" defaultMessage="Email is required." />
+                        ) : isInvalidEmail(user.email) ? (
+                          <FormattedMessage id="editUserDialog.invalidEmail" defaultMessage="Email is invalid." />
```
Comment (rart) [inline ui/app/src/components/EditUserDialog/EditUserDialogUI.tsx]:
```suggestion
                          <FormattedMessage id="editUserDialog.emailRequired" defaultMessage="Email is required" />
                        ) : isInvalidEmail(user.email) ? (
                          <FormattedMessage id="editUserDialog.invalidEmail" defaultMessage="Email is invalid" />
```

File: ui/app/src/components/EditUserDialog/EditUserDialogUI.tsx [hunk 5]
```diff
@@ -190,11 +212,22 @@ export function EditUserDialogUI(props: EditUserDialogUIProps) {
...
                       id="lastName"
                       onChange={(e) => onInputChange({ lastName: e.currentTarget.value })}
                       value={user.lastName}
                       fullWidth
+                      error={validateRequiredField(user.lastName) || validateFieldMinLength('lastName', user.lastName)}
+                      helperText={
+                        validateRequiredField(user.lastName) ? (
+                          <FormattedMessage
+                            id="editUserDialog.lastNameRequired"
+                            defaultMessage="Last Name is required."
```
Comment (rart) [inline ui/app/src/components/EditUserDialog/EditUserDialogUI.tsx]:
```suggestion
                            defaultMessage="Last Name is required"
```

File: ui/app/src/components/EditUserDialog/EditUserDialogUI.tsx [hunk 6]
```diff
@@ -173,11 +182,24 @@ export function EditUserDialogUI(props: EditUserDialogUIProps) {
...
                       value={user.firstName}
                       fullWidth
+                      error={
+                        validateRequiredField(user.firstName) || validateFieldMinLength('firstName', user.firstName)
+                      }
+                      helperText={
+                        validateRequiredField(user.firstName) ? (
+                          <FormattedMessage
+                            id="editUserDialog.firstNameRequired"
+                            defaultMessage="First Name is required."
```
Comment (rart) [inline ui/app/src/components/EditUserDialog/EditUserDialogUI.tsx]:
```suggestion
                            defaultMessage="First Name is required"
```

File: ui/app/src/components/EditUserDialog/EditUserDialogUI.tsx [hunk 7]
```diff
@@ -64,6 +65,10 @@ const translations = defineMessages({
   confirmCancel: {
     id: 'words.no',
     defaultMessage: 'No'
+  },
+  invalidMinLength: {
+    id: 'userInfoDialog.invalidMinLength',
+    defaultMessage: 'Min {length} characters.'
```
Comment (rart) [inline ui/app/src/components/EditUserDialog/EditUserDialogUI.tsx]:
```suggestion
    defaultMessage: 'Min {length} characters'
```

File: ui/app/src/components/EditUserDialog/EditUserDialogContainer.tsx [hunk 8]
```diff
@@ -209,6 +235,9 @@ export function EditUserDialogContainer(props: EditUserDialogContainerProps) {
       onEnableChange={onEnableChange}
       onCancelForm={onCancelForm}
       onResetPassword={onResetPassword}
+      isInvalidEmail={isInvalidEmail}
+      validateRequiredField={validateRequiredField}
+      validateFieldMinLength={validateFieldMinLength}
```
Comment (rart) [inline ui/app/src/components/EditUserDialog/EditUserDialogContainer.tsx]: avoid utils as props

File: ui/app/src/components/EditGroupDialog/EditGroupDialogContainer.tsx [hunk 9]
```diff
@@ -212,12 +225,15 @@ export function EditGroupDialogContainer(props: EditGroupDialogContainerProps) {
...
       onChangeValue={onChangeValue}
+      submitOk={submitOk}
       onSave={onSave}
       onCancel={onCancel}
       onAddMembers={onAddMembers}
       onRemoveMembers={onRemoveMembers}
       inProgressIds={inProgressIds}
       isDirty={isDirty}
+      validateRequiredField={validateRequiredField}
+      validateGroupNameMinLength={validateGroupNameMinLength}
```
Comment (rart) [inline ui/app/src/components/EditGroupDialog/EditGroupDialogContainer.tsx]: Avoid utils as props

**Comment by jvega190** [comment]
@rart I moved the maxLength props for userDialogs to UserManagement/utils.ts for it to be consistent. I also added the maxLength restrictions to EditUserDialog (it was only added to CreateUserDialog)

**Review by rart — APPROVED** [review]

**Pull request merged**